proposal/color-4-new-spaces-js.d.ts.md

```diff
@@ -0,0 +1,864 @@
+    * [`toSpace`](#tospace)
+    * [`isLegacy`](#islegacy)
+    * [`isInGamut`](#isingamut)
+    * [`toGamut`](#togamut)
+    * [`channelsOrNull`](#channelsornull)
+    * [`channels`](#channels)
+    * [`channel`](#channel)
+    * [`alpha`](#alpha)
+    * [`isChannelMissing`](#ischannelmissing)
+    * [`isAlphaMissing`](#isalphamissing)
+    * [`isChannelPowerless`](#ischannelpowerless)
+    * [`interpolate`](#interpolate)
+  * [Updated Color Functions](#updated-color-functions)
+    * [`change`](#change)
+  * [New Constructors](#new-constructors)
+    * [LAB Channel Constructor](#lab-channel-constructor)
+    * [LCH Channel Constructor](#lch-channel-constructor)
+    * [Predefined RGB Channel Constructor](#predefined-rgb-channel-constructor)
+    * [XYZ Channel Constructor](#xyz-channel-constructor)
+    * [Legacy Color Constructors with `space`](#legacy-color-constructors-with-space)
+  * [Deprecations](#deprecations)
+* [Procedures](#procedures)
+  * [Parsing a Channel Value](#parsing-a-channel-value)
+  * [Changing a Component Value](#changing-a-component-value)
+* [Embedded Protocol](#embedded-protocol)
+  * [SassColor](#sasscolor)
+  * [Removed SassScript values](#removed-sassscript-values)
+
+## API
+
+```ts
+import {Value} from '../spec/js-api/value';
+```
+
+## Types
+
+### Color Space Definitions
+
+```ts
+
+export type ColorSpaceHSL = 'hsl';
+
+export type ChannelNameHSL = 'hue' | 'saturation' | 'lightness';
+
+export type ColorSpaceHWB = 'hwb';
+
+export type ChannelNameHWB = 'hue' | 'whiteness' | 'blackness';
+
+export type ColorSpaceLAB = 'lab' | 'oklab';
+
+export type ChannelNameLAB = 'lightness' | 'a' | 'b';
+
+export type ColorSpaceLCH = 'lch' | 'oklch';
+
+export type ChannelNameLCH = 'lightness' | 'chroma' | 'hue';
+
+export type ColorSpaceRGB =
+  | 'a98-rgb'
+  | 'display-p3'
+  | 'prophoto-rgb'
+  | 'rgb'
+  | 'srgb'
+  | 'srgb-linear';
+
+export type ChannelNameRGB = 'red' | 'green' | 'blue';
+
+export type ColorSpaceXYZ = 'xyz' | 'xyz-d50' | 'xyz-d65';
+
+export type ChannelNameXYZ = 'x' | 'y' | 'z';
+
+export type ChannelName =
+  | ChannelNameHSL
+  | ChannelNameHWB
+  | ChannelNameLAB
+  | ChannelNameLCH
+  | ChannelNameRGB
+  | ChannelNameXYZ;
+
+export type KnownColorSpace =
+  | ColorSpaceHSL
+  | ColorSpaceHWB
+  | ColorSpaceLAB
+  | ColorSpaceLCH
+  | ColorSpaceRGB
+  | ColorSpaceXYZ;
+
+export type PolarColorSpace = ColorSpaceHSL | ColorSpaceHWB | ColorSpaceLCH;
+
+export type RectangularColorSpace = Omit<KnownColorSpace, PolarColorSpace>;
+
+export type HueInterpolationMethod =
+  | 'decreasing'
+  | 'increasing'
+  | 'longer'
+  | 'shorter';
+```
+
+### New Color Functions
+
+```ts
+export class SassColor extends Value {
+```
+
+#### `space`
+
+Returns the name of [`internal`]'s space.
+
+[`internal`]: ../spec/js-api/value/color.d.ts.md#internal
+
+```ts
+get space(): KnownColorSpace;
+```
+
+#### `toSpace`
+
+Returns the result of [`color.to-space(internal, space)`] as a new SassColor.
```

Comment:
I don't think you need to explicitly specify "as a new SassColor" here, particularly since it's likely some implementations will want to optimize the "same as existing space" case to return the same object.

Reply:
In light of your comment on `channel` (and to avoid a potentially non-insignificant difference between implementations),  I went further and specified that it should be a no-op if the space is not changing.  Changed in be40def4295de5511b6d2870742d877c68e982d9,